src/core/directory_reader.cpp

```diff
@@ -0,0 +1,17 @@
+#include "directory_reader.hpp"
+
+#include <filesystem>
+
+namespace octo::core
+{
+DirectoryReader::DirectoryReader(std::string_view path)
+{
+    path_ = path;
```

Comment:
read about member initializer lists here https://www.learncpp.com/cpp-tutorial/constructor-member-initializer-lists/

and try to update according to that article

Reply:
also, you can check out tokenizer.cpp